resteasy-jaxrs/src/main/java/org/jboss/resteasy/microprofile/config/ResteasyConfig.java

```diff
@@ -21,23 +28,50 @@
       SERVLET_CONTEXT
    }
 
-   public ResteasyConfig()
+   static {
+      Method getConfig;
+      Method getOptionalValue;
+      Method getValue;
+      Method getPropertyNames;
+      try {
+          final ClassLoader classLoader = getClassLoader();
+          Class.forName("org.jboss.resteasy.microprofile.config.ServletConfigSourceImpl", false, classLoader);
+          final Class<?> configProvider = Class.forName("org.eclipse.microprofile.config.ConfigProvider", false, classLoader);
+          getConfig = configProvider.getDeclaredMethod("getConfig", ClassLoader.class);
+          final Class<?> config = Class.forName("org.eclipse.microprofile.config.Config", false, classLoader);
+          getOptionalValue = config.getDeclaredMethod("getOptionalValue", String.class, Class.class);
+          getValue = config.getDeclaredMethod("getValue", String.class, Class.class);
+          getPropertyNames = config.getDeclaredMethod("getPropertyNames");
+          MP_CONFIG_PRESENT = true;
+      } catch (Throwable ignore) {
+          getConfig = null;
+          getOptionalValue = null;
+          getValue = null;
+          getPropertyNames = null;
+      }
+      GET_CONFIG = getConfig;
+      GET_OPTIONAL_VALUE = getOptionalValue;
+      GET_VALUE = getValue;
+      GET_PROPERTY_NAMES = getPropertyNames;
+  }
+
+   @SuppressWarnings("unchecked")
+   public String getValue(String propertyName)
    {
-      try
+      if (!MP_CONFIG_PRESENT)
       {
-         Class.forName("org.eclipse.microprofile.config.spi.ConfigSource");
-         Class.forName("org.jboss.resteasy.microprofile.config.ServletConfigSourceImpl");
-         config = ResteasyConfigProvider.getConfig();
+         return null;
       }
-      catch (Throwable e)
+      try
       {
-         // Leave config == null.
+         final Object config = GET_CONFIG.invoke(null, getClassLoader());
+         Optional<String> opt = (Optional<String>) GET_OPTIONAL_VALUE.invoke(config, propertyName, String.class);
+         return opt.orElse(null);
       }
-   }
-
-   public String getValue(String propertyName)
-   {
-      return config == null ? null : config.getOptionalValue(propertyName, String.class).orElse(null);
+      catch (IllegalAccessException | InvocationTargetException e) {
+         LogMessages.LOGGER.debugf(e, "Failed to invoke the configuration API method %s.", GET_VALUE);
```

Comment:
The argument should be `GET_OPTIONAL_VALUE`.

Reply:
CopyAndPasteException ...